content/guides/end-to-end-testing/amazon-cognito-authentication.md

```diff
@@ -7,22 +7,23 @@ e2eSpecific: true
 
 ## <Icon name="graduation-cap"></Icon> What you'll learn
 
-- Programmatically authenticate with
-  [Amazon Cognito](https://aws.amazon.com/cognito) via a custom Cypress command
-- Adapting your [Amazon Cognito](https://aws.amazon.com/cognito) application for
-  programmatic authentication during testing
+- Authenticate with [`cy.origin()`](/api/commands/origin) or programmatically
+  (legacy) to [Amazon Cognito](https://aws.amazon.com/cognito) via a custom
+  Cypress command
+- If needed, adapting your [Amazon Cognito](https://aws.amazon.com/cognito)
+  application for programmatic authentication during testing
 
 </Alert>
 
 <Alert type="success">
 
-<strong class="alert-header">Why authenticate programmatically?</strong>
+<strong class="alert-header">Authenticate by visiting a different domain with
+[`cy.origin()`](/api/commands/origin)</strong>
 
-Typically, logging in a user within your app by authenticating via a third-party
-provider requires visiting login pages hosted on a different domain. Since each
-Cypress test is limited to visiting domains of the same origin, we can subvert
-visiting and testing third-party login pages by programmatically interacting
-with the third-party authentication API to login a user.
+Programmatic authentication for Cognito is now considered a legacy
```

Comment:
I think it's good to explain why we are recommending origin first
```suggestion
Typically, logging in a user within your app by authenticating via a third-party provider requires visiting login pages hosted on a different domain. Before Cypress 12, Cypress tests were limited to visiting domains of the same origin, so programatic login was the only option for authenticating with a third-party authentication API. 
```

Reply:
Took a stab at this in e14bf7170